implementations/rust/ockam/ockam_command/src/node/delete.rs

```diff
@@ -40,32 +40,90 @@ impl DeleteCommand {
+            all_nodes,
+        ))
     } else {
-        "Are you sure you want to delete this node?"
+        DeleteMode::Single(cmd.node_name)
     };
-    if opts
-        .terminal
-        .confirmed_with_flag_or_prompt(cmd.yes, prompt_msg)?
-    {
-        if cmd.all {
-            delete_all_nodes(&opts, cmd.force)?;
-            opts.terminal
-                .stdout()
-                .plain(fmt_ok!("All nodes have been deleted"))
-                .write_line()?;
-        } else {
-            delete_node(&opts, &node_name, cmd.force)?;
-            opts.terminal
-                .stdout()
-                .plain(fmt_ok!("Node with name '{}' has been deleted", &node_name))
-                .machine(&node_name)
-                .json(serde_json::json!({ "node": { "name": &node_name } }))
-                .write_line()?;
+
+    match delete_mode {
+        DeleteMode::All => {
+            if opts.terminal.confirmed_with_flag_or_prompt(
+                cmd.yes,
+                "Are you sure you want to delete all nodes?",
+            )? {
+                delete_all_nodes(&opts, cmd.force)?;
+                opts.terminal
+                    .stdout()
+                    .plain(fmt_ok!("All nodes have been deleted"))
+                    .write_line()?;
+            }
         }
-    }
+        DeleteMode::Single(cmd_node_name) => {
+            if opts.terminal.confirmed_with_flag_or_prompt(
+                cmd.yes,
+                "Are you sure you want to delete this node?",
+            )? {
+                let node_name = get_node_name(&opts.state, &cmd_node_name);
+                delete_node(&opts, &node_name, cmd.force)?;
+                opts.terminal
+                    .stdout()
+                    .plain(fmt_ok!("Node with name '{}' has been deleted", &node_name))
+                    .machine(&node_name)
+                    .json(serde_json::json!({ "node": { "name": &node_name } }))
+                    .write_line()?;
+            }
+        }
+        DeleteMode::Selected(option_selected_list) => {
+            let selected_list = option_selected_list.unwrap();
+            if selected_list.is_empty() {
+                opts.terminal
+                    .stdout()
+                    .plain("No nodes selected for deletion")
+                    .write_line()?;
+                return Ok(());
+            }
+
+            if opts
+                .terminal
+                .confirm_interactively(format!(
+                    "Would you like to delete these items : {:?}?",
+                    selected_list
+                ))
+                .unwrap_or(false)
+            {
+                let (deleted_nodes, delete_fail_nodes_with_error) =
+                    delete_selected_nodes(&opts, selected_list, cmd.force)?;
+
+                for deleted_node in deleted_nodes {
+                    println!("✅ Deleted Node: '{}'", &deleted_node);
```

Comment:
All the `println` calls should be replaced by calls to `opts.terminal.write_line`. Also, you should use `fmt_ok` or the other [fmt helpers](https://github.com/build-trust/ockam/blob/dcae9ee9a2a343a13beaf78d29b0efd66e04555b/implementations/rust/ockam/ockam_command/src/terminal/fmt.rs) instead of embedding emojis directly.